### PR TITLE
Update default helm image for helm-upgrade-from- tasks

### PR DESCRIPTION
**Pull request by mikecroft**
# Changes

This PR updates the default image used in the `helm-upgrade-from-repo` and `helm-upgrade-from-source` tasks from [v3.6.0](https://hub.docker.com/layers/lachlanevenson/k8s-helm/v3.6.0/images/sha256-b3510a2fb35cad3631e245e416b5e8b1d03b0184e0d125a4797e055201852ba3?context=explore) to [v3.10.2](https://hub.docker.com/layers/lachlanevenson/k8s-helm/v3.10.2/images/sha256-4ea644ab5330f22864ca12bde02414360d371de9db25f0c177638fa81bc69908?context=explore).

The main motivation for this change is that the existing version does not include [the `--dependency-upgrade` option](https://github.com/helm/helm/pull/9659). This change therefore provides the ability for charts to _not_ package their dependencies.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations][authoring]
- [ ] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [ ] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [ ] Commit messages follow [commit message best practices][commit]
- [ ] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/helm-upgrade-from-repo/0.3/helm-upgrade-from-repo.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# This Task will do a helm upgrade based on the given helm repo and chart
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: helm-upgrade-from-repo
+  labels:
+    app.kubernetes.io/version: "0.2"
```
Comment (fhopfensperger) [inline task/helm-upgrade-from-repo/0.3/helm-upgrade-from-repo.yaml:7]:
```suggestion
    app.kubernetes.io/version: "0.3"
```

File: task/helm-upgrade-from-repo/0.3/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Install the Task
+
+#### Workspaces
+
+* **source**: A [Workspace](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md) volume containing the helm chart.
+
+### helm install / upgrade from repo
+
+```
+kubectl apply -f https://api.hub.tekton.dev/v1/resource/tekton/task/helm-upgrade-from-repo/0.2/raw
```
Comment (fhopfensperger) [inline task/helm-upgrade-from-repo/0.3/README.md:14]:
```suggestion
kubectl apply -f https://api.hub.tekton.dev/v1/resource/tekton/task/helm-upgrade-from-repo/0.3/raw
```

File: task/helm-upgrade-from-source/0.4/helm-upgrade-from-source.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: helm-upgrade-from-source
+  labels:
+    app.kubernetes.io/version: "0.3"
```
Comment (fhopfensperger) [inline task/helm-upgrade-from-source/0.4/helm-upgrade-from-source.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.4"
```

**Comment by mikecroft** [comment]
Is there anything else I need to do to get this merged? It looks like it only needs approval, but I'm happy to add more changes if needed.

**Comment by fhopfensperger** [comment]
Thank you very much for the PR.
I was off the last weeks.
/lgtm

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**